test/test_backend/routes/test_graph.py

```diff
@@ -0,0 +1,402 @@
+"""Tests
+
+How to run:
+    python -m unittest discover
+
+TODO: for big JSON, save without indent
+todo: Siggie already wrote some code that gets the commit hash, because we show that in TermHub's UI. i can probably
+ re-use that here for DRYness.
+"""
+import asyncio
+import csv
+import json
+import os
+import subprocess
+import sys
+import unittest
+from io import StringIO
+from pathlib import Path
+from typing import Dict, List, Set, Tuple
+
+from networkx import DiGraph
+
+THIS_DIR = Path(os.path.dirname(__file__))
+PROJECT_ROOT = THIS_DIR.parent.parent.parent
+sys.path.insert(0, str(PROJECT_ROOT))
+# noinspection PyUnresolvedReferences rel_graph_exists_just_not_if_name_eq_main
+from backend.routes.graph import concept_graph, concept_graph, REL_GRAPH, get_missing_in_between_nodes
+
+THIS_STATIC_DIR = THIS_DIR / 'static'
+STATIC_DIR_concept_graph = THIS_STATIC_DIR / 'concept_graph'
+# remove last two codeset_ids from many-small because not like the others (include rxnorm)
+# many small,many-small,"1000002363, 1000002657, 1000007602, 1000013397, 1000010688, 1000015307, 1000031299",45
+
+# cardiomyopathies, expect cid 619077 to be missing
+TEST_CASES_FAST = """
+testType,testName,codeset_ids,timeoutSeconds,hide_vocabs
+small,cardiomyopathies,35275316,45,Nebraska Lexicon
+single 2000,autoimmune 1,101398605,180
+many small,many-small,"1000002363, 1000002657, 1000007602, 1000013397, 1000010688",45,Nebraska Lexicon
+mixed 6000 to 21000,Sulfonylureas,"417730759, 423850600, 966671711, 577774492",120
+mixed 30 to 3000,autoimmune 2,"101398605, 947369784, 287650725, 283328624, 115052941",240
+single 30000,antibiotics 1,909552172,180
+single small,single-small,1000002363,30
+single small,single-small-second-time,1000002363,30
+"""
+TEST_CASES_HOLD = """
+testType,testName,codeset_ids,timeoutSeconds
+"""
+TEST_CASES = TEST_CASES_FAST
+EXTRA_CONCEPT_IDS = []
+HIDE_VOCABS = ['RxNorm Extension']
+
+
+class AsyncTests(unittest.TestCase):
```

Comment:
## Got `async` tests to work
They wouldn't run with the normal test suite. I had to write this wrapper class. [ChatGPT helped](https://chatgpt.com/c/d9859ecb-c919-44bd-b8f5-8e8c9f09e2f6).